shell/platform/android/test/io/flutter/embedding/engine/systemchannels/PlatformChannelTest.java

```diff
@@ -113,6 +115,61 @@ public void setSystemExclusionRectsSendsJSONExceptionOnIncorrectDataShape() thro
         );
     }
 
+    @Test
+    public void getSystemExclusionRectsSendsExclusionRectsToFrameworkOnSuccess() throws JSONException {
+        DartExecutor dartExecutor = mock(DartExecutor.class);
+        PlatformChannel platformChannel = new PlatformChannel(dartExecutor);
+        PlatformMessageHandler platformMessageHandler = mock(PlatformMessageHandler.class);
+        platformChannel.setPlatformMessageHandler(platformMessageHandler);
+
+        int top = 0;
+        int right = 500;
+        int bottom = 250;
+        int left = 0;
+
+        ArrayList<Rect> expectedExclusionRects = new ArrayList<Rect>();
+        Rect gestureRect = new Rect(left, top, right, bottom);
+        expectedExclusionRects.add(gestureRect);
+        when(platformMessageHandler.getSystemGestureExclusionRects()).thenReturn(expectedExclusionRects);
```

Comment:
One stylistic recommendation that I have is to define the input and output up here at the top during the setup phase. It can often help readability. So above this comment you have the fake input (I think). Down at the bottom of the test you assemble the expected output. You might consider defining the expected output right here, just below the fake input. This allows readers to immediately see the fake input and expected output, together, without reading the rest of the test.